pkg/cassandra/config.go

```diff
@@ -3,153 +3,99 @@ package cassandra
+	// TODO postprocess cassandra.yaml settings, e.g. mountable volumes or resource.Quantity syntax
+	if len(config.CassandraYaml) > 0 {
+		out["cassandra-yaml"] = config.CassandraYaml
 	}
-	addNumTokens(apiConfig, cfg)
-	addEncryptionOptions(apiConfig, cfg)
-	handleDeprecatedJvmOptions(cfg)
-	if out, err := preMarshalConfig(reflect.ValueOf(cfg), apiConfig.ServerVersion, string(apiConfig.ServerType)); err != nil {
-		return nil, err
-	} else {
-		return json.Marshal(out)
+
+	// dse.yaml is an unstructured map, we simply append it to the output
+	// TODO validate dse.yaml settings with JSON schemas
+	// TODO postprocess dse.yaml settings, e.g. mountable volumes or resource.Quantity syntax
+	if len(config.DseYaml) > 0 {
+		out["dse-yaml"] = config.DseYaml
 	}
-}
 
-// cassConfig is an internal type that is semantically equivalent to an api.CassandraConfig object,
-// with the addition of a few hidden fields that the user cannot set directly.
-type cassConfig struct {
-	*api.CassandraConfig   `cass-config:";recurse"`
-	StartRpc               bool              `cass-config:"^3.11.x:cassandra-yaml/start_rpc;retainzero"`
-	ServerEncryptionStores *encryptionStores `cass-config:"cassandra-yaml/server_encryption_options;recurse"`
-	ClientEncryptionStores *encryptionStores `cass-config:"cassandra-yaml/client_encryption_options;recurse"`
-}
+	// JvmOptions is a struct, we need to convert it to a map using preMarshalConfig
+	jvmOptionsVal := reflect.ValueOf(config.JvmOptions)
+	jvmOptionsOut, err := preMarshalConfig(jvmOptionsVal, serverVersion, serverType)
+	if err != nil {
+		return nil, err
+	}
+	out.PutAll(jvmOptionsOut)
 
-type encryptionStores struct {
-	Keystore           string `cass-config:"keystore"`
-	KeystorePassword   string `cass-config:"keystore_password"`
-	Truststore         string `cass-config:"truststore"`
-	TruststorePassword string `cass-config:"truststore_password"`
+	return json.Marshal(out)
 }
 
-func addNumTokens(template *DatacenterConfig, cfg *cassConfig) {
+func addNumTokens(template *DatacenterConfig) {
 	// Even though we default to Cassandra's stock defaults for num_tokens, we need to
 	// explicitly set it because the config builder defaults to num_tokens: 1
-	if cfg.CassandraYaml.NumTokens == nil {
-		version := template.ServerVersion
-		if template.ServerType == api.ServerDistributionCassandra && version.Major() == 3 {
-			cfg.CassandraYaml.NumTokens = pointer.Int(256)
-		} else {
-			cfg.CassandraYaml.NumTokens = pointer.Int(16)
-		}
+	// FIXME when importing existing clusters, we should not override the user's num_tokens setting – even if it's unset (1)
+	if template.ServerType == api.ServerDistributionCassandra && template.ServerVersion.Major() == 3 {
+		template.CassandraConfig.CassandraYaml.PutIfAbsent("num_tokens", 256)
+	} else {
+		template.CassandraConfig.CassandraYaml.PutIfAbsent("num_tokens", 16)
 	}
 }
 
-func addEncryptionOptions(template *DatacenterConfig, cfg *cassConfig) {
```

Comment:
Moved these to another existing function called `handleEncryptionOptions` declared in encryption.go. I think it's easier if these 2 functions are merged together.